selfdrive/camerad/cameras/camera_common.cc

```diff
@@ -79,37 +87,20 @@ void CameraBuf::init(cl_device_id device_id, cl_context context, CameraState *s,
   }};
   yuv_transform = ci->bayer ? transform_scale_buffer(transform, db_s) : transform;
 
-  for (int i = 0; i < UI_BUF_COUNT; i++) {
-    VisionImg img = visionimg_alloc_rgb24(device_id, context, rgb_width, rgb_height, &rgb_bufs[i]);
-    if (i == 0) {
-      rgb_stride = img.stride;
-    }
-  }
-  tbuffer_init(&ui_tb, UI_BUF_COUNT, name);
-  tbuffer_init2(&camera_tb, frame_buf_count, "frame", relase_callback, s);
-
-  // yuv back for recording and orbd
-  pool_init(&yuv_pool, YUV_COUNT);
-  yuv_tb = pool_get_tbuffer(&yuv_pool);
-
-  yuv_width = rgb_width;
-  yuv_height = rgb_height;
-  yuv_buf_size = rgb_width * rgb_height * 3 / 2;
-
-  for (int i = 0; i < YUV_COUNT; i++) {
-    yuv_ion[i] = visionbuf_allocate_cl(yuv_buf_size, device_id, context);
-    yuv_bufs[i].y = (uint8_t *)yuv_ion[i].addr;
-    yuv_bufs[i].u = yuv_bufs[i].y + (yuv_width * yuv_height);
-    yuv_bufs[i].v = yuv_bufs[i].u + (yuv_width / 2 * yuv_height / 2);
-  }
+  vipc_server->create_buffers(rgb_type, UI_BUF_COUNT, true, rgb_width, rgb_height);
+  rgb_stride = vipc_server->get_buffer(rgb_type)->stride;
+
+  vipc_server->create_buffers(yuv_type, YUV_COUNT, false, rgb_width, rgb_height);
 
   if (ci->bayer) {
     cl_program prg_debayer = build_debayer_program(device_id, context, ci, this);
     krnl_debayer = CL_CHECK_ERR(clCreateKernel(prg_debayer, "debayer10", &err));
     CL_CHECK(clReleaseProgram(prg_debayer));
+  } else {
+    krnl_debayer = nullptr;
```

Comment:
`krnl_debayer` has already been initialized as nullptr by `CameraBuf() = default;`  .maybe we can remove the `else` here